packages/merge-styles/src/mergeStyleSets.ts

```diff
@@ -88,7 +99,15 @@ export function mergeStyleSets(...styleSets: Array<IStyleSet | undefined | false
  * @param styleSets - One or more style sets to be merged.
  */
 export function mergeStyleSets(...styleSets: Array<IStyleSet | undefined | false | null>): IProcessedStyleSet<any> {
-  return mergeCssSets(styleSets as any, getStyleOptions());
+  let shadowConfig: ShadowConfig | undefined = undefined;
+  let sets = styleSets;
+  const first = styleSets[0];
+  if (first && first.hasOwnProperty && first.hasOwnProperty('stylesheetKey')) {
+    shadowConfig = styleSets[0] as ShadowConfig;
+    sets = styleSets.slice(1);
+  }
```

Comment:
As I understand it, to be able to have the shadow DOM part work correctly, the user needs to add the stylesheetkey themselves? like so:
 ```tsx
const classNames = mergeStyleSets({
  foo: {
    stylesheetKey: '__global__',
    marginTop: -20,
  },
  bar: {
    padding: '1px 20px',
  },
});
```

This opens up to errors like adding the stylesheetKey to the second set and therefore not handling shadow DOM correctly, right?

Reply:
I actually need to update the type for this function to include `ShadowConfig` :) And I should probably make that test more exhaustive.

A couple things:

1. If no shadow DOM config is provided the implementation will fallback to using the "global" key for you. Users are never required to pass a stylesheet key.
2. Only the first object can have a `stylesheetKey` and it applies to all styles passed in the call. So in your example all the styles would end up in the "global" bucket. There is a problem with your example though in that the `foo` styles would be omitted. I think this is okay because you'd have to opt in to what you're doing here so it's something you would account for in the change.

Expanding further on point 1 we need to provide this fallback for two reasons. First, this is a new, optional feature so we need to provide a default. Second, as a practical matter we cannot opt everything into shadow DOM in a nice way as not all calls to `merge-styles` APIs are within Fluent. An example of this is `initializeIcons()`: it calls `merge-styles` APIs outside of Fluent controls (yes, it is a Fluent package that we should update to play nice with shadow DOM, but I've left it for now because it's a good example of something outside the scope of Fluent React UI controls). In these "outside of Fluent cases" we need a code path to collect all the styles so they can participate in shadow DOM.

Reply:
Ah, that makes sense. Thanks for the explanation!